01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+
+```{r}
+#| label: query-ipums-api
+#| cache: true
+#| eval: false
+## 2.1. ACS: keep households living not living in gq
+
+# Run this the first time you run this code to avoid requesting the same data
+# multiple times
+
+acs_vars <- c("YEAR", "SAMPLE", "SERIAL", "CBSERIAL", "HHWT", "CLUSTER", "state", "county", "MET2013", "CITY", "PUMA", "STRATA", "GQ", "OWNERSHP", "OWNERSHPD", "VALUEH", "PERNUM", "PERWT", "RACE", "RACED", "HISPAN", "HISPAND", "AGE")
+
+years <- 2014:2022
+# usa_ext_def <- define_extract_usa(
+#   description = "ACS 1 Year 2014-2022",
+#   samples = paste0("us", years, "a"),
+#   variables = acs_vars
+# )
+#
+# usa_ext_submitted <- submit_extract(usa_ext_def)
+# usa_ext_complete <- wait_for_extract(usa_ext_submitted)
+#
+# # download_extract() will return a character vector including the file paths
+# # to all downloaded files.
+#
+# filepath <- download_extract(usa_ext_complete,
+#   download_dir = here("01_financial-well-being/data")
+# )
+```
+
+### Clean data {#sec-clean-data}
+
+We load the 2014-2022 1 Year ACS microdata we downloaded from IPUMS above. IPUMS loads the data as a person-level data set with some household variables.Therefore, we can analyze the data at the household level by keeping the first person sampled in the household.
+
+```{r}
+#| label: load-acs
+#| cache: true
+#| dependson: "import-acs-one-year"
+#| include: false
+
+# Set microdata to whatever the file you downloaded in the previous chunk is called
+microdata <- "usa_00010.xml"
+
+acs1yr_raw <-
+  read_ipums_micro(here(glue("01_financial-well-being/data/{microdata}"))) |>
+  janitor::clean_names() |>
+  rename(
+    ownership = ownershp,
+    state = statefip,
+    county = countyfip,
+  ) |>
+  filter(
+    year != 2020,
+    pernum == 1
+  )
+```
+
+The [ipumsr](https://tech.popdata.org/ipumsr/articles/value-labels.html#remove-unused-value-labels) documentation says that:
+
+> "labelled vectors are not intended for use throughout the entire analysis process. Instead, they should be used during the initial data preparation process to convert raw data into values that are more meaningful. These can then be converted to other variable types (often factors) for analysis. Unfortunately, this isn’t a process that can typically be automated, as it depends primarily on the research questions the data will be used to address. However, ipumsr provides several functions to manipulate value labels to make this process easier.
+
+A useful way of finding missing value labels is to call `ipums_var_info(acs1yr)` and inspect the `val_labels`.
+
+Notice that missing values for "value of housing units" (`valueh`) are labelled as 9999998, and 9999999. We fix this before doing any computations.
+
+```{r}
+#| label: handle-missingness
+
+
+acs1yr <-
+  acs1yr_raw |>
+  mutate(
+    valueh = lbl_na_if(valueh, ~ .lbl %in% c("Missing", "N/A")),
+  )
+
+acs1yr |>
+  assertr::verify(valueh < 9999998 | is.na(valueh))
```

Comment:
It took a bit to understand how this handles the missing values exactly, from a code logic perspective (since the function lbl_na_if comes from ipumsr (https://rdrr.io/cran/ipumsr/man/lbl_na_if.html), just noting this.